Mathlib/Algebra/GroupPower/Lemmas.lean

```diff
@@ -2,35 +2,1316 @@
+    C = 0 ∨ 0 < C ∧ 0 ≤ r := by
+  have : 0 ≤ C := by simpa only [pow_zero, mul_one] using h 0
+  refine' this.eq_or_lt.elim (fun h => Or.inl h.symm) fun hC => Or.inr ⟨hC, _⟩
+  refine' nonneg_of_mul_nonneg_right _ hC
+  simpa only [pow_one] using h 1
+#align sign_cases_of_C_mul_pow_nonneg sign_cases_of_C_mul_pow_nonneg
+
+end LinearOrderedSemiring
+
+section LinearOrderedRing
+
+variable [LinearOrderedRing R] {a : R} {n : ℕ}
+
+@[simp]
+theorem abs_pow (a : R) (n : ℕ) : |a ^ n| = |a| ^ n :=
+  (pow_abs a n).symm
+#align abs_pow abs_pow
+
+section bit1
+
+set_option linter.deprecated false
+
+@[simp]
+theorem pow_bit1_neg_iff : a ^ bit1 n < 0 ↔ a < 0 :=
+  ⟨fun h => not_le.1 fun h' => not_le.2 h <| pow_nonneg h' _, fun ha => pow_bit1_neg ha n⟩
+#align pow_bit1_neg_iff pow_bit1_neg_iff
+
+@[simp]
+theorem pow_bit1_nonneg_iff : 0 ≤ a ^ bit1 n ↔ 0 ≤ a :=
+  le_iff_le_iff_lt_iff_lt.2 pow_bit1_neg_iff
+#align pow_bit1_nonneg_iff pow_bit1_nonneg_iff
+
+@[simp]
+theorem pow_bit1_nonpos_iff : a ^ bit1 n ≤ 0 ↔ a ≤ 0 := by
+  simp only [le_iff_lt_or_eq, pow_bit1_neg_iff]
+  refine' ⟨_, _⟩
+  · rintro (hpos | hz)
+    · left
+      exact hpos
+    · right
+      exact (pow_eq_zero_iff'.1 hz).1
+  · rintro (hneg | hz)
+    · left
+      exact hneg
+    · right
+      simp [hz, bit1]
+#align pow_bit1_nonpos_iff pow_bit1_nonpos_iff
+
+@[simp]
+theorem pow_bit1_pos_iff : 0 < a ^ bit1 n ↔ 0 < a :=
+  lt_iff_lt_of_le_iff_le pow_bit1_nonpos_iff
+#align pow_bit1_pos_iff pow_bit1_pos_iff
+
+theorem strict_mono_pow_bit1 (n : ℕ) : StrictMono fun a : R => a ^ bit1 n := by
+  intro a b hab
+  cases' le_total a 0 with ha ha
+  · cases' le_or_lt b 0 with hb hb
+    · rw [← neg_lt_neg_iff, ← neg_pow_bit1, ← neg_pow_bit1]
+      exact pow_lt_pow_of_lt_left (neg_lt_neg hab) (neg_nonneg.2 hb) (bit1_pos (zero_le n))
+    · exact (pow_bit1_nonpos_iff.2 ha).trans_lt (pow_bit1_pos_iff.2 hb)
+  · exact pow_lt_pow_of_lt_left hab ha (bit1_pos (zero_le n))
+#align strict_mono_pow_bit1 strict_mono_pow_bit1
+
+end bit1
+
+/-- Bernoulli's inequality for `n : ℕ`, `-2 ≤ a`. -/
+theorem one_add_mul_le_pow (H : -2 ≤ a) (n : ℕ) : 1 + (n : R) * a ≤ (1 + a) ^ n :=
+  one_add_mul_le_pow' (mul_self_nonneg _) (mul_self_nonneg _) (neg_le_iff_add_nonneg'.1 H) _
+#align one_add_mul_le_pow one_add_mul_le_pow
+
+/-- Bernoulli's inequality reformulated to estimate `a^n`. -/
+theorem one_add_mul_sub_le_pow (H : -1 ≤ a) (n : ℕ) : 1 + (n : R) * (a - 1) ≤ a ^ n := by
+  have : -2 ≤ a - 1 := by
+    rwa [← one_add_one_eq_two, neg_add, ← sub_eq_add_neg, sub_le_sub_iff_right]
+  simpa only [add_sub_cancel'_right] using one_add_mul_le_pow this n
+#align one_add_mul_sub_le_pow one_add_mul_sub_le_pow
+
+end LinearOrderedRing
+
+namespace Int
+
+#check Nat.cast_pow
+
+/- Porting note: removed `@[simp]` because of linter error:
+
+Left-hand side simplifies from
+  ↑(Int.natAbs x ^ 2)
+to
+  ↑(Int.natAbs x) ^ 2
+using
+  simp only [Nat.cast_pow]
+Try to change the left-hand side to the simplified term!
```

Comment:
If the simplifier doesn't like it, why not also add the simp lemma which it does like, with the left hand side changed as it suggests, and tag that `simp` instead?

Reply:
Done. Took a shortcut and reduced the new lemma to the old one, though there is presumably a shorter proof.